runhouse/resources/hardware/utils.py

```diff
@@ -711,3 +752,92 @@ def __str__(self):
         if self.retry:
             s += ", retry in {0}ms".format(self.retry)
         return s
+
+
+###################################
+# KUBERNETES SETUP
+###################################
+def setup_kubernetes(
+    namespace: Optional[str] = None,
+    kube_config_path: Optional[str] = None,
+    context: Optional[str] = None,
+    **kwargs,
+):
+    if kwargs.get("provider") and not kwargs.get("provider") == "kubernetes":
+        raise ValueError(
+            f"Recieved non kubernetes provider {kwargs.get('provider')} with kubernetes specific "
+            "cluster arguments."
+        )
+
+    if (
+        kwargs.get("server_connection_type")
+        and kwargs.get("server_connection_type") != ServerConnectionType.SSH
+    ):
+        raise ValueError(
+            "Runhouse K8s Cluster server connection type must be set to `ssh`. "
+            f"You passed {kwargs.getserver_connection_type}."
+        )
+
+    if context and namespace:
+        logger.warning(
+            "You passed both a context and a namespace. Ensure your namespace matches the one in your context.",
+        )
+
+    launcher = kwargs.get("launcher") or configs.launcher
+    if namespace and launcher == "local":
+        # Set the context only if launching locally
+        # check if user passed a user-defined namespace
+        cmd = f"kubectl config set-context --current --namespace={namespace}"
+        try:
+            process = subprocess.run(
+                cmd,
+                shell=True,
+                check=True,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                text=True,
+            )
+            logger.debug(process.stdout)
+            logger.info(f"Kubernetes namespace set to {namespace}")
+
+        except subprocess.CalledProcessError as e:
+            logger.info(f"Error: {e}")
+
+    # Q - should the following also only be set up for local launcher?
```

Comment:
we set up something similar in the launcher too, it's slightly different because the kube config is in a temp dir and not in `~/.kube`, so I think better to only have this be for the local flow